allennlp/modules/time_distributed.py

```diff
@@ -15,28 +17,55 @@ class TimeDistributed(torch.nn.Module):
 
     Note that while the above gives shapes with ``batch_size`` first, this ``Module`` also works if
     ``batch_size`` is second - we always just combine the first two dimensions, then split them.
+
+    It also reshapes keyword arguments unless they are not tensors or their name is specified in
+    the optional ``pass_through`` iterable.
     """
     def __init__(self, module):
-        super(TimeDistributed, self).__init__()
+        super().__init__()
         self._module = module
 
-    def forward(self, *inputs):  # pylint: disable=arguments-differ
-        reshaped_inputs = []
-        for input_tensor in inputs:
-            input_size = input_tensor.size()
-            if len(input_size) <= 2:
-                raise RuntimeError("No dimension to distribute: " + str(input_size))
+    def forward(self, *inputs, pass_through: Optional[Iterable] = None,  # pylint: disable=arguments-differ
+                **kwargs):
+        if pass_through is None:
+            pass_through = []
+
+        reshaped_inputs = [self._reshape_tensor(input_tensor) for input_tensor in inputs]
+
+        # Need some input to then get the batch_size and time_steps.
+        some_input = None
+        if inputs:
+            some_input = inputs[-1]
 
-            # Squash batch_size and time_steps into a single axis; result has shape
-            # (batch_size * time_steps, input_size).
-            squashed_shape = [-1] + [x for x in input_size[2:]]
-            reshaped_inputs.append(input_tensor.contiguous().view(*squashed_shape))
+        reshaped_kwargs = {}
+        ignored_kwargs = {}
+        for key, value in kwargs.items():
+            if key in pass_through or not isinstance(value, torch.Tensor):
+                ignored_kwargs[key] = value
+            else:
+                if some_input is None:
+                    some_input = value
 
-        reshaped_outputs = self._module(*reshaped_inputs)
+                reshaped_kwargs[key] = self._reshape_tensor(value)
+
+        reshaped_outputs = self._module(*reshaped_inputs, **reshaped_kwargs, **ignored_kwargs)
+
+        if some_input is None:
+            raise RuntimeError("No input tensor to time-distribute")
 
         # Now get the output back into the right shape.
-        # (batch_size, time_steps, [hidden_size])
```

Comment:
`hidden_size` name didn't make much sense to me here. `output_size` did. Was there a reason for it?

Reply:
`output_size` seems fine to me.